src/goal.cpp

```diff
@@ -236,40 +236,43 @@ CommandCost CmdSetGoalCompleted(TileIndex tile, DoCommandFlag flags, uint32 p1,
  * @param flags type of operation
  * @param p1 various bitstuffed elements
  * - p1 = (bit  0 - 15) - Unique ID to use for this question.
- * - p1 = (bit 16 - 23) - Company or client for which this question is.
- * - p1 = (bit 24 - 25) - Question type.
- * - p1 = (bit 31) - Question target: 0 - company, 1 - client.
- * @param p2 Buttons of the question.
+ * - p1 = (bit 16 - 31) - Company or client for which this question is.
+ * @param p2 various bitstuffed elements
+ * - p2 = (bit 0 - 17) - Buttons of the question.
+ * - p2 = (bit 29 - 30) - Question type.
+ * - p2 = (bit 31) - Question target: 0 - company, 1 - client.
  * @param text Text of the question.
  * @return the cost of this operation or an error
  */
 CommandCost CmdGoalQuestion(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
 {
 	uint16 uniqueid = (GoalType)GB(p1, 0, 16);
 	CompanyID company = (CompanyID)GB(p1, 16, 8);
-	ClientIndex client = (ClientIndex)GB(p1, 16, 8);
-	byte type = GB(p1, 24, 2);
-	bool is_client = HasBit(p1, 31);
+	ClientID client = (ClientID)GB(p1, 16, 16);
```

Comment:
If `ClientIndex` is the "proper" type here (as discussed previously), why the change to `ClientID`? I'm not a fan of using arbitrary values for enum types, especially potentially well beyond their defined ranges, as certain aggressive gcc optimisations can truncate enum types or skip range checks...

Reply:
If I'm understanding the issue right, the problem is that `ClientIndex` is a local value and only `ClientID` is in fact synced across the network.